python/plugins/processing/core/Processing.py

```diff
@@ -107,6 +111,40 @@ def deinitialize():
             QgsApplication.processingRegistry().removeProvider(p)
 
         Processing.BASIC_PROVIDERS = []
+        Processing.REGISTERED_PARAMETERS = dict()
+
+    @staticmethod
+    def registerParameter(id, name, parameter, metadata=dict(), description=None, exposeToModeller=True):
```

Comment:
Can we move this to the C++ processing registry  please, and add unit tests for it?